src/searchModal.js

```diff
@@ -98,16 +119,106 @@ export default function searchModalFactory(config) {
+                //set up the inner resource selector
+                selectionMode: 'single',
+                selectClass: true,
+                classUri: rootClassUri,
+                showContext: false,
+                showSelection: false
+            });
+
+            // when a class query is triggered, update selector options with received resources
+            resourceSelector.on('query', function (params) {
+                if (instance.config.testMocks.classTree) {
+                    resourceSelector.update(instance.config.testMocks.classTree, params);
+                    return;
+                }
+                params.classOnly = true;
+                const route = urlUtil.route('getAll', 'RestResource', 'tao');
+                request(route, params)
+                    .then(response => {
+                        resourceSelector.update(response.resources, params);
+                    })
+                    .catch(e => instance.trigger('error', e));
+            });
+
+            /*
+             * the first time selector opions are updated the root class is selected. Promise is
+             * resolved so init process continues only when class input value has been set
+             */
+            resourceSelector.on('update', function () {
+                resourceSelector.off('update');
+
+                resourceSelector.select(initialClassUri);
+                resolve();
+            });
+
+            // then new class is selected, set its label into class filter input and hide filter container
+            resourceSelector.on('change', function (selectedValue) {
+                classFilterInput.val(_.map(selectedValue, 'label')[0]);
+                classTreeContainer.hide();
+            });
+
+            setResourceSelectorUIBehaviour();
+        });
+    }
+
     /**
      * Inits template selectors and sets initial search query on search input
      */
     function initUiSelectors() {
         searchButton = $('.btn-search', instance.getElement());
         clearButton = $('.btn-clear', instance.getElement());
-        searchInput = $('.search-bar-container input', instance.getElement());
+        searchInput = $('.generic-search-input', instance.getElement());
+        classFilterInput = $('.class-filter', instance.getElement());
+        classTreeContainer = $('.class-tree', instance.getElement());
+        classFilterContainer = $('.class-filter-container', instance.getElement());
         searchButton.on('click', search);
         clearButton.on('click', clear);
-        searchInput.val(config.query);
+        searchInput.val(
+            instance.config.criterias && instance.config.criterias.search ? instance.config.criterias.search : ''
+        );
+        // remove class filter if it is not enabled
+        if (classFilterIsEnabled === false) {
+            classFilterContainer.remove();
+        }
```

Comment:
We shouldn't create container if we're going to remove it on initialization. Please, move this logic to the template